### ISSUE TITLE
Twisted.Quotes does not serve its original intended purpose

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2723]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#2723|
|Type|enhancement|
|Created|2007-06-25 20:53:41Z|
|Branch|https://github.com/twisted/twisted/tree/relevant-quotes-2723|

# Problem

The idea for a Twisted quotefile started with a tradition at Origin, where each released game included a quotefile from the game's development team.  This allowed the users to get a feel for the developers' culture, and the developers to feel a bit more connected to the users by sharing it.

At the time I originally thought of having a quote file, I assumed that the Twisted team would simply grow and the team would acquire new members over time; there was no "final release" so one quotefile that just got bigger made sense.

However, the current quotefile has several problems:

  * The default sort order when reading it in an editor or pager - newest quotes last - presents users first with the quotations least relevant to the current release of Twisted.
  * The quotefile is huge, and so readers are overwhelmed by its length and don't read it as often as they used to.  (I am guessing this just from the amount of off-the-cuff feedback I used to get about the quotefile's utility and the feedback I get now about its relative incomprehensibility.)
  * The quotefile includes lots of quotes from people who don't even work on Twisted any more (and in many cases, never worked on Twisted).  Those quotes should't be removed, since they were perhaps relevant to the hackers' state of mind at the time, but again... not relevant today.

# Proposed Solution

The quotefile for a given Twisted release should reflect the culture that produced that release.  The team around Twisted changes with the ebb and flow of different developers' free time, so I think a good solution here would be to split the quotefile up into several different files, and only include one with each release.

If we ever manage to fix the problem that releases take a really long time, we might want to aggregate them differently, but I think for the time being one file / one release makes sense.

<details><summary>Searchable metadata</summary>

```
trac-id__2723 2723
type__enhancement enhancement
reporter__glyph glyph
priority__lowest lowest
milestone__ 
branch__branches_relevant_quotes_2723 branches/relevant-quotes-2723
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__fun__documentation fun, documentation
time__1182804821000000 1182804821000000
changetime__1336517413000000 1336517413000000
version__None None
owner__glyph glyph
cc__thijs cc__jesstess cc__khorn
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

exarkun complained -- and rightly so, I think -- that this ticket described a potential solution without really describing a problem.  I've completely replaced both the summary and the description to reflect what I really meant when I filed it.

Lots of tickets in Twisted (and other trackers, for that matter) suffer from this problem, and I hope that this ticket can serve as a useful example for how to better describe tickets in the future.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @glyph<br>@thijstriemstra set status to `new`|
|-|-|

Not sure where to start..

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @thijstriemstra|
|-|-|

There are two possible parts to this solution.

The first, easy part, would be to move the current Twisted.Quotes to doc/historic/quotes/legacy.quotes, and add a step to the release process where we move Twisted.Quotes to doc/historic/quotes/<release>.quotes.

We'd keep putting new quotes into the root quotes file, and choose a QOTR out of that.

Of more historical interest would be to actually look at the timeline of commits to legacy.quotes and break it up into files for each release, i.e. doc/historic/quotes/1.0.quotes, 1.3.quotes, 2.0.quotes, etc.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

So there's a news generator now with files in ```topfiles``` describing a change to twisted. I guess such a .quote file would go there as well, generated by something similar to that changelog tool?

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_7"></a>|@khorn commented|
|-|-|

thijs:

I actually came to this ticket just now intending to suggest exactly that.

You beat me to it.

+1

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst commented|
|-|-|

A minimalist solution would be to reverse the sort order, so newer quotes are at the top.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph commented|
|-|-|

(In [34178]) Branching to 'relevant-quotes-2723'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_11"></a>|@glyph commented|
|-|-|

Replying to [itamar](#note_9):
> A minimalist solution would be to reverse the sort order, so newer quotes are at the top.

Okay I implemented the "maximalist" solution instead, before this ticket turns into a kafka-esque parody of how our development process prevents anyone from ever getting anything done.  (Really?  "svn mv" is so much work that we have to start bikeshedding about alternative work-reducing approaches?)  I also added [the relevant instruction to the release process here](http://twistedmatrix.com/trac/wiki/ReleaseProcess?action=diff&version=149).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_12"></a>|@itamarst commented|
|-|-|

I still think we should reverse the chronological order on the quotes files. The current branch seems like only half a solution: irrelevant/confusing/embarrassingly juvenile quotes are still the first ones you read if you want to look at more than one quote.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph commented|
|-|-|

Replying to [itamar](#note_12):
> I still think we should reverse the chronological order on the quotes files. The current branch seems like only half a solution: irrelevant/confusing/embarrassingly juvenile quotes are still the first ones you read if you want to look at more than one quote.

I left the original quotefile as-is in order to allow a spelunker to more easily get 'svn blame' to allow them to annotate quotes by actual date if they so desire, but I have no objection to adding new quotes at the top of the current file.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_14"></a>|@itamarst set owner to @itamarst|
|-|-|

I don't think this is good enough, so I guess I'll do some more work myself.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_15"></a>|@glyph removed owner|
|-|-|

Replying to [itamar](#note_14):
> I don't think this is good enough, so I guess I'll do some more work myself.

Don't think this is good enough for what?  Is it an improvement or not?  Does it address the problem or not?  If there's more to do, please file separate tickets.  We don't have to fix everything at once here.  This comment is not clear enough to constitute a review.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_16"></a>|@itamarst set owner to @glyph|
|-|-|

Yeah, sorry about that. I'll open separate ticket for splitting it up by year. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_17"></a>|@glyph set status to `closed`|
|-|-|

(In [34356]) Merge relevant-quotes-2723.

Author: glyph

Reviewer: itamar

Fixes: [#2723](https://github.com/twisted/twisted/issues/2723)

This cycles the quotefile into a new location, so that quotes are relevant to
the current release.
